turbinia/workers/analysis/macho.py

```diff
@@ -0,0 +1,600 @@
+      file_name (str): file name of the fat binary.
+    Returns:
+      FatBinary: the parsed Mach-O Fat Binary details.
+    """
+    result.log(f'---------- start fat binary ------------')
+    macho_fd.seek(0)
+    data = macho_fd.read()
+    hashes = Hashes()
+    hashes.md5 = self._GetDigest(md5.MD5Hasher(), data)
+    hashes.sha256 = self._GetDigest(sha256.SHA256Hasher(), data)
+    hashes.tlsh = tlsh.hash(data)
+    hashes.ssdeep = pyssdeep.get_hash_buffer(data)
+    parsed_fat_binary = ParsedFatBinary(hashes)
+    parsed_fat_binary.entropy = self._GetDigest(entropy.EntropyHasher(), data)
+    fat_binary_stats = os.stat(macho_path)
+    parsed_fat_binary.size = fat_binary_stats.st_size
+    return parsed_fat_binary
+
+  def _ParseMachoBinary(self, macho_fd, evidence, binary, result, file_name):
+    """Parses a Mach-O binary.
+    Args:
+      macho_fd (int): file descriptor to the fat binary.
+      evidence (Evidence object):  The evidence to process
+      binary (lief.MachO.Binary): binary to be parsed.
+      result (TurbiniaTaskResult): The object to place task results into.
+      filename (str): file name of the binary.
+    Returns:
+      ParsedBinary: the parsed binary details.
+    """
+    result.log(f'------------ start binary --------------')
+    fat_offset = binary.fat_offset
+    binary_size = binary.original_size
+    macho_fd.seek(fat_offset)
+    data = macho_fd.read(binary_size)
+    hashes = Hashes()
+    hashes.md5 = self._GetDigest(md5.MD5Hasher(), data)
+    hashes.sha256 = self._GetDigest(sha256.SHA256Hasher(), data)
+    hashes.symhash = self._GetSymhash(binary)
+    hashes.tlsh = tlsh.hash(data)
+    hashes.ssdeep = pyssdeep.get_hash_buffer(data)
+    parsed_binary = ParsedBinary(
+        hashes=hashes, segments=None, symbols=None, imports=None, flags=None)
+    parsed_binary.entropy = self._GetDigest(entropy.EntropyHasher(), data)
+    parsed_binary.size = binary_size
+    parsed_binary.fat_offset = fat_offset
+    parsed_binary.magic = hex(binary.header.magic.value)
+    parsed_binary.segments = self._GetSegments(binary, result)
+    parsed_binary.symbols = self._GetSymbols(binary, result)
+    flags = []
+    for flag in binary.header.flags_list:
+      flags.append(str(flag).split(".")[-1])
+    parsed_binary.flags = flags
+    imports = []
+    for lib in binary.libraries:
+      imp = Import()
+      imp.name = lib.name
+      imp.size = hex(lib.size)
+      imp.offset = hex(lib.command_offset)
+      imports.append(imp)
+    parsed_binary.imports = imports
+
+    if binary.has_code_signature:
+      try:
+        # we went knee-deep into asn1 parsing to get our hands on the signature details.
+        # it is very easy to get this wrong and shoot ourselves in the foot.
+        # so for the time being we want to monitor to see how we went and catch any exception if needed.
+        # once we have confidence that we got it right we can narrow down the catching of the exceptions.
```

Comment:
Good catch. This `try` should be removed now as if it fails it should actually fail. In case the binary has a `code_signature` section the parsing has to work or it deserves an exception to be thrown.